Fix #4195: Preserve texture transforms when applying PBR materials

Aqil-Ahmad opened the pull request:
## Description
Fixes texture transforms being reset when switching from Blinn-Phong to PBR materials via drag-and-drop. Previously, custom scale, offset, and rotation settings would be lost, making it tedious to switch between PBR materials while maintaining UV mapping.
The fix preserves existing texture entry transforms and applies them to all PBR texture channels when switching materials, but only when non-default transforms are present to avoid unnecessary overrides.

## Related Issues

Issue Link: closes #4195

---

## Checklist

Please ensure the following before requesting review:

- [x] I have provided a clear title and detailed description for this pull request.
- [ ] If useful, I have included media such as screenshots and video to show off my changes.
- [x] The PR is linked to a relevant issue with sufficient context.
- [x] I have tested the changes locally and verified they work as intended.
- [ ] All new and existing tests pass.
- [x] Code follows the project's style guidelines.
- [x] Documentation has been updated if needed.
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have reviewed the [contributing guidelines](https://github.com/secondlife/viewer/blob/develop/CONTRIBUTING.md).

---

## Additional Notes

Aqil-Ahmad commented:
I have read the CLA Document and I hereby sign the CLA

akleshchev commented on indra/newview/llselectmgr.cpp:
```diff
@@ -1996,9 +1996,55 @@ bool LLSelectMgr::selectionSetGLTFMaterial(const LLUUID& mat_id)
+                // Check if we have non-default texture entry transforms to preserve
+                tep->getScale(&existing_scale_s, &existing_scale_t);
+                tep->getOffset(&existing_offset_s, &existing_offset_t);
+                existing_rotation = tep->getRotation();
```
This looks wrong. I don't think material offsets are stored there, they should be in override material.

Aqil-Ahmad commented:
@akleshchev Thanks for all the detailed feedback! I've made the changes based on your suggestions. Let me know if you spot any other issues - really appreciate the thorough review!

akleshchev reviewed (changes requested):
Scenario:
1. Rez a cube
2. Apply non default rotation, scale to texture
3. Apply a PBR material
Observe A: PBR material inherits scale from texture (I'm fairly sure request meant preservation of values between strcitly PBR, not texture->PBR, but it isn't really a problem here and arguably better this way, as long as lindens are aware of the change)
4. Change PBR rotation and scale to be default ones
5. Apply a new PBR material
Observe B: material inherits values from step 2 instead of values from step 4.

Expected: switching one material with another material should inherit values from the material, not from texture.

Aqil-Ahmad commented:
> Scenario:
> 
> 1. Rez a cube
> 2. Apply non default rotation, scale to texture
> 3. Apply a PBR material
>    Observe A: PBR material inherits scale from texture (I'm fairly sure request meant preservation of values between strcitly PBR, not texture->PBR, but it isn't really a problem here and arguably better this way, as long as lindens are aware of the change)
> 4. Change PBR rotation and scale to be default ones
> 5. Apply a new PBR material
>    Observe B: material inherits values from step 2 instead of values from step 4.
> 
> Expected: switching one material with another material should inherit values from the material, not from texture.

Fixed. Updated code now preserves transforms from the current material type (PBR transforms when switching between PBR materials, texture transforms only when switching from texture to PBR)

akleshchev reviewed (changes requested):
Scenario:
1. Rez an object
2. Assign a texture that makes it easier to see offsets
3. Assign Horizontal scale 1.5 or vertical offset to 0.4
<img width="1133" height="900" alt="image" src="https://github.com/user-attachments/assets/272aa7e2-b8cd-4d6e-8314-2a21682803bf" />
4. Drop a pbr material on top
<img width="1070" height="858" alt="image" src="https://github.com/user-attachments/assets/83a337da-6485-42b1-b00e-29dcb131cf45" />

Observe: there is mismatch from texture

There is difference between how texture and pbr values work despite values being identical (as I understand it, it's expected, textures are 'legacy', while PBR should be format compliant). Unfortunately they can not work identically, you will need to translate texture scale/offset to pbr scale/offset.

bennettgoble commented:
Closed as stale. @Aqil-Ahmad if you wish to re-open with all the PR could you include example media and ensure tests run? Thanks!

akleshchev commented on indra/newview/llselectmgr.cpp:
```diff
@@ -1996,9 +1996,53 @@ bool LLSelectMgr::selectionSetGLTFMaterial(const LLUUID& mat_id)
+                
+                // Queue the override to be applied
+                LLGLTFMaterialList::queueModify(objectp, te, override_material);
+            }
```
Correction: not queueModify but queueApply
